meta-client/src/main/java/org/apache/helix/metaclient/api/ConnectStateChangeListener.java

```diff
@@ -0,0 +1,36 @@
+package org.apache.helix.metaclient.api;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+public interface ConnectStateChangeListener {
+  /**
+   * Called when the connection state has changed.
+   */
+  void handleConnectStateChanged(MetaClientInterface.ConnectState currentState) throws Exception;
+  /**
+   * Called when new connection is established
+   */
+  void handleNewConnection(final String sessionId) throws Exception;
+  /**
+   * Called when new connection failed to established
+   */
+  void handleConnectionEstablishmentError(final Throwable error) throws Exception;
```

Comment:
ZkMetaClient will use a realAwareZkClient. The ZK stateChangeListsner throws a generic exception. If we catch and throw another kind of exception in metaClient, then it may break existing user behavior.

Reply:
Ok, that makes sense. Thanks.